### PR TITLE
Simplify stock_threshold_alert field in UpdateProductCommand and UpdateCombinationCommand

### LINKED ISSUE #30750: Handle low_stock_alert field by low_stock_threshold value in product and combination handlers
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

Currently in new product page combination form data handler the low_stock_alert is depending on dedicated disabling_switch_low_stock_threshold input (and there is a dedicated $lowStockAlert property in the UpdateCombinationCommand which handles dedicated low_stock_alert field in database). 

### Describe the solution you'd like

We discussed with @jolelievre that it will probably be better/cleaner to handle the field based on the low_stock_threshold value, so basically if the value is = 0 then we assume the low_stock_alert = false.

### Alternatives you've considered

_No response_

### Additional context

_No response_

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | https://github.com/PrestaShop/PrestaShop/issues/30750
| Type?             | refacto
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | See bellow
| Fixed ticket?     | Closes https://github.com/PrestaShop/PrestaShop/issues/30750
| Related PRs       | If theme, autoupgrade or other module change is needed to make this change work, provide a link to related PRs here.
| Sponsor company   | Your company or customer's name goes here (if applicable).

**How to test**
In product page v2 edit product stock tab - make sure that low stock alert switch and low stock level input works as intended (when alert switch is on then you can input the low stock level value, else you cannot and the threshold is not updated)
Same with combination form and combination bulk edition form.

### CODE REVIEW

**Comment by zuk3975** [comment]
@jolelievre do you think I should also remove the $lowStockAlert prop from queries like GetProductForEditing and GetCombinationForEditing etc and only use threshold value.. ? If I do that, then we should probably deprecate the "low_stock_alert" field in object models and replace all its usages to comply with low_stock_threshold instead. Not sure if we want to do that now 🤔

**Review by matks — APPROVED** [review]

**Review by SharakPL — APPROVED** [review]

**Review by florine2623 — COMMENTED** [review]
Hello @zuk3975 ,

Automated tests link : https://github.com/florine2623/testing_pr/actions/runs/3968186979

What about this comment ? Should this be tested here or in another PR ?
-> https://github.com/PrestaShop/PrestaShop/pull/30936#issuecomment-1396762064

Tested the following :
- combination form
- combination bulk edition form
- checked for multiple combinations in products (size + color + paper type)
- check database `ps_product_shop`

The _combination form_ and _combination bulk edition form_ work well. I can switch on and off the low stock alert for 1, multiple and all combinations. ✅ 

I just noticed this thing when I do the following steps :
- Edit low stock alert with bulk action > Save
- Click edit any combination with classic combination form
- Edit the low stock alert once again > Save
- See a number displayed
Screenshot 2023-01-20 at 15 22 28


https://user-images.githubusercontent.com/16019289/213721384-75e5776e-b344-496d-b19e-d41a5702ce41.mov

I couldn't reproduce the issue on develop without this PR, the feature did not exist yet.

Also, whether I change the low stock alert or not, nothing changes in the database, in ps_product_shop.
I always have `low_stock_threshold` = NULL and `low_stock_zero` = 0.
I don't know if that's the expected behavior. Can you let me know ? ^^

Thanks!

**Comment by zuk3975** [comment]
1. @florine2623 thanks. The number in the history footer seems to be the combination id (the one thats being edited), I agree it seems a bit strange, but I see it was intentionally added in another [PR here ](https://github.com/PrestaShop/PrestaShop/commit/6d84c0be74fd92ca4d7536492a3d6e6a9a621622#diff-2ae182f8458ad00306a2dcf059f58d8f6839e3c80c66b071fe4a5a4ba394d91cR67) so it is unrelated with the PR

2. product_shop `low_stock_threshold` and `low_stock_alert` values seems to change correctly. One note though - when you edit product with combinations its only changing in `product_attribute_shop`, when changing product without ocmbinations then it should change in `product_shop`.

So besides these issues, then its a QA ✅ ?

**Review by florine2623 — COMMENTED** [review]
Hello @zuk3975 ,

1. I think it should be validated first by PM. I think it's strange that a number appears without explanation on the page. 
@PrestaShop/product-team , it that supposed to be displayed here ?
213721862-67f65a79-ca82-4844-9506-90dd3058e9c8


2. Ok, thanks for the information. I confirm this behavior.
After I modify my low stock alert for combination product, the database is updated in `ps_product_attribute_shop` for `low_stock_threshold` column
This is QA ✅ 

Thanks ^^

**Comment by jolelievre** [comment]
Hi @florine2623 

I agree that this extra number makes no sense, and it as added in a PR that migrated Vue2 to Vue3 so it should have had no impact on features, so I guess it was added by mistake. I think it should be removed but I'm not sure it should block this PR so I see two solutions:
- fix this in a dedicated PR (the change as not added in this PR anyway it's already present on develop) to avoid blocking this one
- it can be fixed in this PR but then the next QA should be quick and only focus on the removal of this extra text

@MatShir what do you think?

**Comment by MatShir** [comment]
Let's create an issue and go further with this PR :)

**Review by florine2623 — APPROVED** [review]
Thanks @zuk3975 @jolelievre & @MatShir 

It is QA ✅ then 💪

**Comment by florine2623** [comment]
Here's the issue : https://github.com/PrestaShop/PrestaShop/issues/31007

**Comment by jolelievre** [comment]
Thanks @zuk3975 and @florine2623

File: tests/Integration/Behaviour/Features/Scenario/Product/update_stock.feature [hunk 1]
```diff
@@ -4,7 +4,7 @@
 @clear-cache-before-feature
 @reboot-kernel-before-feature
 @update-stock
-@update-stock-classic
+@update-stock-cl./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s product --tags update-stockassic
```
Comment (zuk3975) [inline tests/Integration/Behaviour/Features/Scenario/Product/update_stock.feature:7]: oops

**Pull request merged**